src/cacheRefresher.js

```diff
@@ -0,0 +1,185 @@
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+(function () {
+    "use strict";
+
+    const { DomUtil } = require("./domUtil");
+    const XtkCaster = require('./xtkCaster.js').XtkCaster;
+    const { Cache } = require('./cache.js');
+
+    /**
+     * @private
+     * @class
+     * @constructor
+     * @memberof Campaign
+     */
+    class MetadataCache extends Cache {
+
+        /**
+         * A in-memory cache for properties values. Not intended to be used directly,
+         * but an internal cache for the Campaign.Client object
+         * 
+         * Cached object are made of
+         * - the key is the propertie name
+         * - the value is a string
+         * 
+         * @param {Storage} storage is an optional Storage object, such as localStorage or sessionStorage
+         * @param {string} rootKey is an optional root key to use for the storage object
+         * @param {number} ttl is the TTL for objects in ms. Defaults to 5 mins
+         */
+        constructor(storage, rootKey, ttl) {
+            super(storage, rootKey, ttl);
+        }
+
+        /**
+         * Cache a properties and its value
+         * 
+         * @param {string} name is the propertie name
+         * @param {string} rawValue string value
+         */
+        put(name, rawValue) {
+            return super.put(name, { value: rawValue });
+        }
+
+        /**
+         * Get the value of a propertie
+         * 
+         * @param {string} name the propertie name
+         * @returns {*} the value
+         */
+        get(name) {
+            const option = super.get(name);
+            return option ? option.value : undefined;
+        }
+    }
+
+    class CacheRefresher {
+
+        /**
+        * A class to refresh regulary a Cache every 10 seconds, by sending a query to get the last modified entities
+        * it hould be used in a client.
+        *
+        * @param {Cache} cache is the cache to refresh
+        * @param {Client} client is the ACC API Client.
+        * @param {ConnectionParameters} connectionParameters used to created the client provide as parameter
+        * @param {string} cacheSchema is the schema present in the cache to be refreshed every 10 seconds
+        * @param {string} rootKey is an optional root key to use for the storage object
+        */
+        constructor(cache, client, connectionParameters, cacheSchema, rootKey) {
+
+            this._cache = cache;
+            this._client = client;
+            this._connectionParameters = connectionParameters;
+            this._cacheSchema = cacheSchema;
+
+            this._storage = connectionParameters._options._storage;
+            this._metadataCache = new MetadataCache(this._storage, `${rootKey}.MetadataCache`, connectionParameters._options.optionCacheTTL);
```

Comment:
As discussed last week, this cache should have it's own TTL and not reuse the options cache TTL